org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/MvnProtocolHandlerService.java

```diff
@@ -129,6 +129,19 @@ public InputStream getInputStream() throws IOException {
       String urlSpec = "jar:" + location.toURI() + "!" + subPath;
       return new URL(urlSpec).openStream();
     }
+
+    @Override
+    public long getLastModified() {
+      try {
+        connect();
+      } catch (IOException e) {
+        return 0;
```

Comment:
I think -1 has to be returned for "cant determine last modified"

Reply:
JavaDoc says:
`@return  the date the resource referenced by this {@code URLConnection} was last modified, or 0 if not known.`
Also `0` is what was returned if not overriding the method at all.

Reply:
Alright, then forget about that :-)